src/Nethermind/Nethermind.TxPool/TxFilteringState.cs

```diff
@@ -1,33 +1,25 @@
 // SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
+using System;
 using Nethermind.Core;
 
 namespace Nethermind.TxPool;
 
-public class TxFilteringState
+public class TxFilteringState(Transaction tx, IAccountStateProvider accounts)
 {
-    private readonly IAccountStateProvider _accounts;
-    private readonly Transaction _tx;
+    private AccountStruct _senderAccount = AccountStruct.TotallyEmpty;
 
-    public TxFilteringState(Transaction tx, IAccountStateProvider accounts)
-    {
-        _accounts = accounts;
-        _tx = tx;
-    }
-
-    private AccountStruct? _senderAccount = null;
     public AccountStruct SenderAccount
     {
         get
         {
-            if (_senderAccount is null)
+            if (_senderAccount.IsTotallyEmpty)
```

Comment:
Transactions cannot be realisticly sent from totally empty accounts, so if it is this should fail on first filter that accesses the account and we shouldn't look it up multiple times.

Reply:
Is it `TotallyEmpty` or is it `default` before its been looked up? `TotallyEmpty` is a valid state for a looked up account whereas `default` isn't.

e.g. CodeHash is empty string hash and StorageHash is empty trie hash

Reply:
i.e. it should transition `default` =>`TotallyEmpty` on first failed lookup

Reply:
Downside its its now a 128 byte compare to test if empty rather than an 8 byte reference check